packages/tooling/fast-tooling/src/web-components/file/file.ts

```diff
@@ -0,0 +1,123 @@
+import { attr, observable } from "@microsoft/fast-element";
+import { isHTMLElement } from "@microsoft/fast-web-utilities";
+import { FileAction } from "../file-action/file-action";
+import { FormAssociatedFile } from "./file.form-associated";
+
+/**
+ * Web component for presenting a file picker dialog and perform some action on the selected files.
+ * The component expects a FileAction component as a child which performs the action.
+ * Example:
+ *   <fast-tooling-file
+ *       accept=".jpg,.jpeg,.gif,.png"
+ *       @change=${x=>x.myChangeHandler}
+ *   >
+ *       Add image
+ *       <fast-tooling-file-action-objecturl role="fileaction" slot="action"></fast-tooling-file-action-objecturl>
+ *   </fast-tooling-file>
+ */
+export class File extends FormAssociatedFile {
+    /**
+     * Reference to the FileAction that will be performed when files are selected.
+     */
+    private fileAction: FileAction;
+
+    /**
+     * The list of file types that will be shown in the file picker.
+     * Matches syntax for the "accept" attribute of the <file type="file"/> element.
+     */
+    @attr
+    public accept: string = "";
+
+    /**
+     * When true the file picker will allow the selection of multiple files.
+     * By default only one file is allowed.
+     */
+    @attr
+    public multiple: boolean = false;
+
+    /**
+     * Callback method for reporting progress of long running FileAction.
+     * The progress value is calculated by the FileAction but should be a number between 0 and 1.
+     */
+    @attr
+    public progressCallback: (progress: number) => void;
+
+    /**
+     * After file(s) are selected this property will contain a string array with the references
+     * to the selected files based on the action (i.e. file name, objectUrl, http url, base64 blob, etc).
+     */
+    @observable
+    public fileReferences: string[] = [];
+
+    /**
+     * Slot reference for actions slot. When children are added to the slot it searches for the first
+     * control with the role="fileaction". Additional actions are ignored.
+     */
+    @observable
+    public action: HTMLSlotElement;
+    private actionChanged(oldValue, newValue): void {
+        if (this.$fastController.isConnected) {
+            if (this.children.length > 0) {
+                Array.from(this.children).every((value: FileAction) => {
```

Comment:
This could be condensed. You shouldn't need to check for the length of this.children. It will be an empty array that you loop through 0 times.

Calling Array.every() will loop through all of the children even after it finds a match. It will return all of these, but you're not assigning them though. Using find will stop once it finds a match so it will be ever so slightly performant.

You can replace lines 60 through 68 with:
this.fileAction = Array.from(this.children).find(this.isFileAction);

Reply:
Updated. However, the original logic would stop after the first match. Array.every() stops if the method returns false.